src/Utils/QtiMarshallerUtil.php

```diff
@@ -93,4 +100,31 @@ public static function marshallValidQti(QtiComponent $component)
 
         return $dom->saveXML($node);
     }
+
+    public function marshallObjectData(QtiComponent $component)
+    {
+        $result = '';
+        $class = new \ReflectionClass(get_class($component));
+        if (property_exists($component, 'data') && property_exists($component, 'type')) {
+            $property = $class->getProperty('data');
+            $property->setAccessible(true);
+            $property_type = $class->getProperty('type');
+            $property_type->setAccessible(true);
+            $type = $property_type->getValue($component);
+            if ($type == 'text/html') {
+                $learnosityServiceObject = ConvertToLearnosityService::getInstance();
+                $inputPath = $learnosityServiceObject->getInputpath();
+                $file = $inputPath . '/' . $property->getValue($component);
+                if (file_exists($file)) {
+                    $result = HtmlExtractorUtil::getHtmlData(realpath($file));
+                } else {
+                    echo 'File not found: ' . $file . PHP_EOL;
```

Comment:
I was thinking about how we can set a better pattern here - I'd rather we not touch `stdout` from inside a functional Util (with the exception of loggers or writers). This could cause unintended side effects if this is re-used elsewhere.

Can we throw an exception instead, and let it be caught and echoed out by one of the calling classes that does allow for side effects?

Reply:
Hi @ttton , I tried to threw an exception for above but then that exception is getting caught in class "convertToLearnosityService" method "convertAssessmentItemInFile" by which it is saying an item is invalid. So I did it by creating a method in class "convertToLearnosityService" . Also logged it.

Reply:
Ok, I see.

Does this kill the operation or does it just skip the item? I'm thinking that is probably what we want if it only renders that item invalid. I can't imagine why you'd want to convert an item if it is missing content, or is incorrectly configured. I think that's something we want to flag as loudly as possible, ideally by refusing to convert it until either the source file is located or the `assessmentItem` is updated with a valid link.

Let me know your thoughts, I'll also raise this internally to see if I can clarify the preferred behaviour.

Reply:
Hello @ttton , it does not kill the process but skip the item saying invalid item. I think it makes sense to skip an item if proper content is not available . 

Let me know if you want me to update it in that way.

Reply:
Hi @pooja31chaudhary, yep agreed. Let's go with skipping the item with an exception.

Reply:
Hi @ttton , I have fixed above issue . Please check latest commit.